crates/hotshot/src/tasks/task_state.rs

```diff
@@ -248,6 +275,23 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> CreateTaskState<TYPES, I>
         handle: &SystemContextHandle<TYPES, I>,
     ) -> QuorumProposalTaskState<TYPES, I> {
         let consensus = handle.hotshot.consensus();
+
+        let event_stream = handle.internal_event_stream.0.clone();
```

Comment:
Same comment as above, looks like this same block occurs a couple times, so I'll stop commenting this. Perhaps it'd be useful as a helper?

Reply:
Yeah, good idea, let me add it as a helper somewhere